app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -1682,42 +1696,79 @@ public void showControlsThenHide() {
         if (DEBUG) {
             Log.d(TAG, "showControlsThenHide() called");
         }
-        showOrHideButtons();
-        showSystemUIPartially();
-
         final int hideTime = binding.playbackControlRoot.isInTouchMode()
-                ? DEFAULT_CONTROLS_HIDE_TIME
-                : DPAD_CONTROLS_HIDE_TIME;
-
-        showHideShadow(true, DEFAULT_CONTROLS_DURATION);
-        animate(binding.playbackControlRoot, true, DEFAULT_CONTROLS_DURATION,
-                AnimationType.ALPHA, 0, () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime));
+            ? DEFAULT_CONTROLS_HIDE_TIME
+            : DPAD_CONTROLS_HIDE_TIME;
+        showControls(
+            DEFAULT_CONTROLS_DURATION,
+            () -> hideControls(DEFAULT_CONTROLS_DURATION, hideTime)
+        );
     }
 
     public void showControls(final long duration) {
+        showControls(duration, null);
+    }
+
+    public void showControls(final long duration, final Runnable callback) {
         if (DEBUG) {
             Log.d(TAG, "showControls() called");
         }
-        showOrHideButtons();
-        showSystemUIPartially();
+
+        final RelativeLayout layoutToShow;
+        if (!isLocked) {
+            showOrHideButtons();
+            showSystemUIPartially();
+            layoutToShow = binding.playbackControlRoot;
+        } else {
+            layoutToShow = binding.lockedRoot;
+        }
+
         controlsVisibilityHandler.removeCallbacksAndMessages(null);
         showHideShadow(true, duration);
-        animate(binding.playbackControlRoot, true, duration);
+        animate(
+            layoutToShow,
+            true,
+            duration,
+            AnimationType.ALPHA,
+            0,
+            callback
```

Comment:
Whats the reason for adding the `callback` parameter and passing it in here?

Reply:
This is a cleanup, showControlsThenHide had duplicated the logic of showControls.